rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -429,14 +430,15 @@ nested_type = '__'.join(type_.namespaced_name())
     char * buffer;
     Py_ssize_t length;
     int rc = PyBytes_AsStringAndSize(encoded_field, &buffer, &length);
-    Py_DECREF(encoded_field);
     if (rc) {
```

Comment:
same about checking exactly for `-1`

Reply:
Checking for not equal zero is common for such error code values imo. Therefore I would leave it as is (also since it is orthogonal to this PR).